ci-operator/step-registry/upi/install/vsphere/upi-install-vsphere-commands.sh

```diff
@@ -0,0 +1,168 @@
+cp -t "${installer_dir}" \
+    ${tfvars_path}
+
+export KUBECONFIG="${installer_dir}/auth/kubeconfig"
+
+function gather_bootstrap_and_fail() {
+    set +e
+    # shellcheck source=/dev/null
+    source "${SHARED_DIR}/govc.sh"
+    # list all the virtual machines in the folder/rp
+    clustervms=$(govc ls "/${GOVC_DATACENTER}/vm/${cluster_name}")
+    for ipath in $clustervms; do
+      # split on /
+      # shellcheck disable=SC2162
+      IFS=/ read -a ipath_array <<< "$ipath";
+      hostname=${ipath_array[-1]}
+
+      # create png of the current console to determine if a virtual machine has a problem
+      echo "$(date -u --rfc-3339=seconds) - capture console image"
+      govc vm.console -vm.ipath="$ipath" -capture "${ARTIFACT_DIR}/${hostname}.png"
+
+      # based on the virtual machine name create variable for each
+      # with ip addresses as the value
+      # wait 1 minute for an ip address to become available
+
+      # shellcheck disable=SC2140
+      declare "${hostname//-/_}_ip"="$(govc vm.ip -wait=1m -vm.ipath="$ipath" | awk -F',' '{print $1}')"
+    done
+
+    # shellcheck disable=SC2154
+    GATHER_BOOTSTRAP_ARGS="${GATHER_BOOTSTRAP_ARGS} --bootstrap ${bootstrap_0_ip}"
+    # shellcheck disable=SC2154
+    GATHER_BOOTSTRAP_ARGS="${GATHER_BOOTSTRAP_ARGS} --master ${control_plane_0_ip} --master ${control_plane_1_ip} --master ${control_plane_2_ip}"
+
+    set -e
+  if test -n "${GATHER_BOOTSTRAP_ARGS}"; then
+    openshift-install --dir=/tmp/artifacts/installer gather bootstrap --key "${SSH_PRIVATE_KEY_PATH}" "${GATHER_BOOTSTRAP_ARGS}"
+  fi
+
+  return 1
+}
+
+function approve_csrs() {
+  # The cluster won't be ready to approve CSR(s) yet anyway
+  sleep 30
+
+  echo "$(date -u --rfc-3339=seconds) - Approving the CSR requests for nodes..."
+  while true; do
+    oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs --no-run-if-empty oc adm certificate approve || true
+    sleep 15
+  done
+}
+
+function update_image_registry() {
+  sleep 30
+
+  echo "$(date -u --rfc-3339=seconds) - Waiting for imageregistry config to be available"
+  while true; do
+    oc get configs.imageregistry.operator.openshift.io/cluster > /dev/null && break
+    sleep 15
+  done
+
+  echo "$(date -u --rfc-3339=seconds) - Patching image registry configuration..."
+  oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState":"Managed","storage":{"emptyDir":{}}}}'
+}
+
+echo "$(date -u --rfc-3339=seconds) - terraform init..."
+terraform init -input=false -no-color &
+wait "$!"
+
+echo "$(date -u --rfc-3339=seconds) - terraform apply..."
+terraform apply -auto-approve -no-color &
+wait "$!"
+
+# The terraform state could be larger than the maximum 1mb
+# in a secret
+tar -Jcf "${SHARED_DIR}/terraform_state.tar.xz" terraform.tfstate
+
+# To ease debugging of ip address use
+cluster_domain=$(<"${SHARED_DIR}"/clusterdomain.txt)
+host -t A "api.${cluster_domain}"
```

Comment:
not sure what this is helping with debugging?

Reply:
I suppose there could be a better way to disable what IP addresses were used but its nice to know what is resolvable. 
The major reason is if/when there is issues when two clusters overlap ip addresses.